contracts/src/Nodes.sol

```diff
@@ -2,149 +2,240 @@
 pragma solidity 0.8.28;
 
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
-
-/**
- * A NFT contract for XMTP Node Operators.
- *
- * The deployer of this contract is responsible for minting NFTs and assigning them to node operators.
- *
- * All nodes on the network periodically check this contract to determine which nodes they should connect to.
- */
-contract Nodes is ERC721, Ownable {
-    constructor() ERC721("XMTP Node Operator", "XMTP") Ownable(msg.sender) {}
-
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/INodes.sol";
+
+
+ /// @title XMTP Node Registry.
+ /// @notice This contract is responsible for minting NFTs and assigning them to node operators.
+ /// Each node is minted as an NFT with a unique ID (starting at 100 and increasing by 100 with each new node).
+ /// In addition to the standard ERC721 functionality, the contract supports node-specific features,
+ /// including node property updates.
+ /// 
+ /// @dev All nodes on the network periodically check this contract to determine which nodes they should connect to.
+ /// The contract owner is responsible for:
+ /// - minting and transferring NFTs to node operators.
+ /// - updating the node operator's HTTP address and MTLS certificate.
+ /// - updating the node operator's minimum monthly fee.
+ /// - updating the node operator's API enabled flag.
+contract Nodes is ERC721, INodes, AccessControl {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    bytes32 public constant NODE_MANAGER_ROLE = keccak256("NODE_MANAGER_ROLE");
+
+    /// @dev The maximum commission percentage that the node operator can receive.
+    uint256 public constant MAX_BPS = 10000;
```

Comment:
Great. One of the asks from our lawyers is to put bounds on any parameters in the contract, so that there is a predictable range for variable fees.